arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/Effect.kt

```diff
@@ -716,17 +777,20 @@ internal class FoldContinuation<R, B>(
   private val error: suspend (Throwable) -> B,
   private val parent: Continuation<B>,
 ) : Continuation<B>, Token(), EffectScope<R> {
-  
+
   constructor(ignored: Token, context: CoroutineContext, parent: Continuation<B>) : this(context, { throw it }, parent)
   constructor(
     ignored: Token,
     context: CoroutineContext,
     error: suspend (Throwable) -> B,
     parent: Continuation<B>,
   ) : this(context, error, parent)
-  
+
   lateinit var recover: suspend (R) -> Any?
-  
+
+  // Add AtomicBoolean to arrow-atomic
+  val isActive: AtomicRef<Boolean> = AtomicRef(true)
```

Comment:
I'm a bit worried about exposing the `AtomicRef` directly. Is it possible to make this `internal`?

Reply:
This is inside an `internal class` so it's `internal` by inheriting the visibility modifier.

Reply:
But it shows in the `arrow-core.api` file... 🤔

Reply:
That's because `internal` is public in the (Java) binary 😭